seirmo/plots/_plot_from_numpy.py

```diff
@@ -0,0 +1,154 @@
+                              Data should has one row per timestep,
+                              and one column for each dependent variable
+        :params:: position: list of integers, gives index of subplot to use
+        :params:: xlabel: str
+        :params:: new_axis: boolean, set to true if data should
+                            be plotted on a second x axis"""
+
+        if len(data_array.shape) == 1:  # Turn any 1D input into 2D
+            if type(times) != np.ndarray or np.sum(np.shape(times)) == 1:
+                # I.e. if only one np.int, or one element array
+                times = np.array(times, ndmin=2)
+                data_array = data_array[np.newaxis, :]
+            else:
+                data_array = data_array[:, np.newaxis]
+
+        assert (
+            times.shape[0] == data_array.shape[0]
+        ), "data and times are not the same length"
+        data_width = data_array.shape[1]  # saves the number of y-var
+
+        assert (
+            position[0] < self._nrows and position[1] < self._ncolumns
+        ), "position and shape are not compatible"
+
+        if new_axis:
+            axis = self._axes[position[0], position[1]].twinx()
+        else:
+            axis = self._axes[position[0], position[1]]
+
+        # formats colour choice if none set - I want to change this
+        # If you change this you are redoing the tests
+        if not colours:
+            colours = plt.cm.viridis(np.linspace(0, 1, data_width))
+
+        # plots the data
+        use_labels = False
+        for i in range(data_width):
+            if i < len(ylabels):
+                use_labels = True
+                axis.plot(times, data_array[:, i],
+                          color=colours[i], label=ylabels[i])
+            else:
+                axis.plot(times, data_array[:, i], color=colours[i])
+        if use_labels:
+            axis.legend()
+        plt.xlabel(xlabel)
+        self._fig.tight_layout()
+        return self._fig, self._axes
+
+    def add_fill(
+        self,
+        times: np.ndarray,
+        ymin: np.ndarray,
+        ymax: np.ndarray,
+        position: list = [0, 0],
+        xlabel: str = "time",
+        ylabel: str = "number of people",
+        colours: str = ["b"],
+        alpha: float = 0.2,
+    ):
+
+        assert (
+            position[0] < self._nrows and position[1] < self._ncolumns
+        ), "position and shape are not compatible"
+        axis = self._axes[position[0], position[1]]
+
+        # plots the data
+        axis.fill_between(
+            times,
+            np.squeeze(ymin),
+            np.squeeze(ymax),
+            color=colours,
+            alpha=alpha,
+            label=ylabel,
+        )
+        axis.legend()
+        plt.xlabel(xlabel)
+        self._fig.tight_layout()
+        return self._fig, self._axes
+
+    def show(self):
+        plt.show()
+
+    def writeToFile(self, filename: str = "SEIR_stochastic_simulation.png"):
```

Comment:
I have not committed this here simply because I wanted to go through and check it wasn't referenced anywhere else (and update the testing), however this has been changed